multiselect_test.go

```diff
@@ -89,17 +92,149 @@ func TestMultiSelectRender(t *testing.T) {
 		},
 	}
 
-	outputBuffer := bytes.NewBufferString("")
-	terminal.Stdout = outputBuffer
-
 	for _, test := range tests {
-		outputBuffer.Reset()
+		r, w, err := os.Pipe()
+		assert.Nil(t, err, test.title)
+
+		test.prompt.WithStdio(terminal.Stdio{Out: w})
 		test.data.MultiSelect = test.prompt
-		err := test.prompt.Render(
+		err = test.prompt.Render(
 			MultiSelectQuestionTemplate,
 			test.data,
 		)
 		assert.Nil(t, err, test.title)
-		assert.Equal(t, test.expected, outputBuffer.String(), test.title)
+
+		w.Close()
+		var buf bytes.Buffer
+		io.Copy(&buf, r)
+
+		assert.Contains(t, buf.String(), test.expected, test.title)
+	}
+}
+
+func TestMultiSelectPrompt(t *testing.T) {
+	tests := []PromptTest{
+		{
+			"Test MultiSelect prompt interaction",
+			&MultiSelect{
+				Message: "What days do you prefer:",
+				Options: []string{"Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"},
+			},
+			func(c *expect.Console) {
+				c.ExpectString("What days do you prefer:  [Use arrows to move, type to filter]")
+				// Select Monday.
+				c.Send("\x1b[B ")
+				c.SendLine("")
+				c.ExpectEOF()
+			},
+			[]string{"Monday"},
+		},
+		{
+			"Test MultiSelect prompt interaction with default",
+			&MultiSelect{
+				Message: "What days do you prefer:",
+				Options: []string{"Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"},
+				Default: []string{"Tuesday", "Thursday"},
+			},
+			func(c *expect.Console) {
+				c.ExpectString("What days do you prefer:  [Use arrows to move, type to filter]")
+				c.SendLine("")
+				c.ExpectEOF()
+			},
+			[]string{"Tuesday", "Thursday"},
+		},
+		{
+			"Test MultiSelect prompt interaction overriding default",
+			&MultiSelect{
+				Message: "What days do you prefer:",
+				Options: []string{"Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"},
+				Default: []string{"Tuesday", "Thursday"},
+			},
+			func(c *expect.Console) {
+				c.ExpectString("What days do you prefer:  [Use arrows to move, type to filter]")
+				// Deselect Tuesday.
+				c.Send("\x1b[B\x1b[B ")
```

Comment:
It should be possible to use the internal `survey/core` methods on a Cursor that points to the `expect.Console`'s stdio to abstract away the actual ansi escape sequences here right? It's not a huge deal but it would definitely make it friendlier for future developers to write these tests if they didn't have to look up these values.

Reply:
Ah darn well that's definitely good enough for now